### PR TITLE
Location Field: Cancel pending requests on new request

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Attempt to resolve network-based race conditions. Adds a new abort controller. Aborts and resets on keystroke.

### CODE REVIEW

File: packages/location-field/src/index.tsx [hunk 1]
```diff
@@ -201,132 +204,128 @@ const LocationField = ({
...
-                const hasResults = parts.length !== 0;
-                const results = hasResults
-                  ? intl.formatList(parts, { type: "conjunction" })
-                  : intl.formatMessage({
-                      description: "Indicates no results",
-                      id: "otpUi.LocationField.noResults"
-                    });
-                const resultsFoundText = intl.formatMessage(
+    getGeocoder(geocoderConfig)
+      // .autocomplete({ text })
```
Comment (daniel-heppner-ibigroup) [inline packages/location-field/src/index.tsx:225]: remove comment

File: packages/location-field/src/index.tsx [hunk 2]
```diff
@@ -185,6 +185,9 @@ const LocationField = ({
   const [isFetching, setFetching] = useState(false);
   const [stateMessage, setMessage] = useState(null);
   const [stateValue, setValue] = useState(getValueFromLocation());
+  const [abortControllers, setAbortController] = useState([
+    new AbortController()
```
Comment (daniel-heppner-ibigroup) [inline packages/location-field/src/index.tsx:189]: Is there any need to intitalize this with an `AbortController`? I don't see this controller being used anywhere. You can set the type of the state with `useState<AbortController[]>`.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Interesting! I have never seen the Abort Controller before. This is a neat solution! I think you may have left some things behind though.

File: packages/location-field/src/index.tsx (hunk 2 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.tsx:189]: Oh thanks for catching this yes this is residue

File: packages/location-field/src/index.tsx [hunk 3]
```diff
@@ -201,132 +202,127 @@ const LocationField = ({
...
-        setFetching(true);
-        setMessage(
-          intl.formatMessage({
-            defaultMessage: "Fetching suggestions…",
-            description:
-              "Hint shown while geocoder suggestions are being fetched",
-            id: "otpUi.LocationField.fetchingSuggestions"
-          })
-        );
+  const geocodeAutocomplete = debounce(300, (text: string) => {
```
Comment (binh-dam-ibigroup) [inline packages/location-field/src/index.tsx:205]: Would have to check in the future if we should consider a `useCallback`.

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving for now. We'll have to revisit the big long `geocodeAutoComplete` method in the future.

File: packages/location-field/src/index.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/location-field/src/index.tsx:205]: I've added a comment

**Pull request merged**